torch/distributed/algorithms/_comm_hooks/allreduce_hook.py

```diff
@@ -0,0 +1,59 @@
+import torch
+import torch.distributed as dist
+from torch.distributed import distributed_c10d
+
+
+class AllReduceState(object):
+    r"""
+    Stores parameters, needed to perform ``all_reduce`` algorithm
```

Comment:
nit:
```suggestion
    Stores state needed to perform the ``all_reduce`` algorithm
```
(just a suggestion)